src/lib/EventCards.svelte

```diff
@@ -20,32 +20,32 @@
   const createDateText = ({
     dateParams: { startDateUTC, endDateUTC, startTimeUTC, endTimeUTC },
   }: CommunityEvent): string => {
-    const startDate = new Date(startDateUTC)
-    const endDate = endDateUTC ? new Date(endDateUTC) : null
-    const startTime = startTimeUTC ? new Date(`2000-01-01T${startTimeUTC}Z`) : null
-    const endTime = endTimeUTC ? new Date(`2000-01-01T${endTimeUTC}Z`) : null
+    const startDate = new Date(`${startDateUTC}T${startTimeUTC ?? '00:00:00'}Z`)
+    const endDate = new Date(`${endDateUTC ?? '2000-01-01'}T${endTimeUTC ?? '00:00:00'}Z`)
 
     const options: Intl.DateTimeFormatOptions = { month: 'long', day: 'numeric', year: 'numeric' }
-    const startDateText = startDate.toLocaleDateString(undefined, options)
-    const endDateText = endDate ? endDate.toLocaleDateString(undefined, options) : null
-    const startTimeText = startTime
-      ? startTime.toLocaleTimeString(undefined, { hour: 'numeric', minute: 'numeric' })
+    const startDateText = startTimeUTC
+      ? startDate.toLocaleDateString('en-US', options)
+      : new Intl.DateTimeFormat('en-US', options).format(startDate)
+    const endDateText = endTimeUTC
+      ? endDate.toLocaleDateString('en-US', options)
+      : new Intl.DateTimeFormat('en-US', options).format(endDate)
+    const startTimeText = startTimeUTC
+      ? startDate.toLocaleTimeString('en-US', { hour: 'numeric', minute: 'numeric' })
       : null
-    const endTimeText = endTime
-      ? endTime.toLocaleTimeString(undefined, { hour: 'numeric', minute: 'numeric' })
+    const endTimeText = endTimeUTC
+      ? endDate.toLocaleTimeString('en-US', { hour: 'numeric', minute: 'numeric' })
```

Comment:
I'm not sure if `en-US` is really the best locale to use for these. AM/PM can be confusing to international users (guilty as charged), so I think the 24-hour clock is the most straight-forward option.

Reply:
I used undefined now, this will result in the local time format being used which should be the best for everyone